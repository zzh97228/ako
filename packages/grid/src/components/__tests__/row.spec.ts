import { mount, VueWrapper } from '@vue/test-utils';
import { Row } from '../row';

describe('row.ts', () => {
  let mountFunc: (options?: object) => VueWrapper<any>;
  beforeEach(() => {
    mountFunc = (options = {}) => {
      return mount(Row, options);
    };
  });

  it('should render row component with customized justify & align props', () => {
    const wrapper = mountFunc({
      props: {
        justify: 'center',
        align: 'end',
      },
    });
    expect(wrapper.classes()).toContain('row');
    expect(wrapper.classes()).toContain('row--justify-content-center');
    expect(wrapper.classes()).toContain('row--align-items-end');
    expect(wrapper.html()).toMatchSnapshot();
  });

  it('should render row component with customized gutter', async () => {
    const wrapper = mountFunc({
      props: {
        gutter: 'xs',
      },
    });
    expect(wrapper.classes()).toContain('row--gutter-xs');
    expect(wrapper.html()).toMatchSnapshot();

    await wrapper.setProps({
      gutter: '4px',
    });
    expect(wrapper.attributes('style')).toContain('margin-left: -4px');
    expect(wrapper.attributes('style')).toContain('margin-right: -4px');
    expect(wrapper.html()).toMatchSnapshot();
  });

  it('should have vertical margin when set \"column\" props', () => {
    const wrapper = mountFunc({
      props: {
        gutter: '8px',
        column: true,
      },
    });
    expect(wrapper.classes()).toContain('row--column');
    expect(wrapper.attributes('style')).toContain('margin-top: -8px');
    expect(wrapper.attributes('style')).toContain('margin-bottom: -8px');
    expect(wrapper.html()).toMatchSnapshot();
  });
});
