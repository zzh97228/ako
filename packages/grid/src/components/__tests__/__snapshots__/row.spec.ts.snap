// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`row.ts should have vertical margin when set "column" props 1`] = `
<div class="row row--column"
     style="margin-top: -8px; margin-bottom: -8px;"
>
</div>
`;

exports[`row.ts should render row component with customized gutter 1`] = `
<div class="row row--gutter-xs">
</div>
`;

exports[`row.ts should render row component with customized gutter 2`] = `
<div class="row"
     style="margin-left: -4px; margin-right: -4px;"
>
</div>
`;

exports[`row.ts should render row component with customized justify & align props 1`] = `
<div class="row row--justify-content-center row--align-items-end">
</div>
`;
